src/main/java/hillelJavaEE_02/pet/PetController.java

```diff
@@ -47,17 +48,17 @@ public String helloWorld() {
     public ResponseEntity<?> getPetById(@PathVariable Integer id) {
         if(id >= pets.size()) {
             return ResponseEntity
-                    //.notFound().build()
-                    .badRequest().body(new ErrorBody("There is no pet with ID = " + id));
+                    .notFound().build();
+                    //.badRequest().body(new ErrorBody("There is no pet with ID = " + id));
         }
 
         return ResponseEntity.ok(pets.get(id));
     }
 
     @PostMapping("/pets")
     public ResponseEntity<Void> createPet(@RequestBody Pet pet) {
-        pets.put(++counter, pet);
-        return ResponseEntity.created(URI.create("/pets/" + counter)).build();
+        pets.put(counter.incrementAndGet(), pet);
+        return ResponseEntity.created(URI.create("/pets/" + counter.get())).build();

```

Comment:
between `counter.incrementAndGet()` and `counter.get()` another thread can change counter :(
use variable for this: `Integer i = counter.incrementAndGet()`